contracts/upgradeable_contracts/BaseBridgeValidators.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity 0.4.24;
+
+import "./Ownable.sol";
+import "../libraries/SafeMath.sol";
+import "../upgradeability/EternalStorage.sol";
+
+
+contract BaseBridgeValidators is EternalStorage, Ownable {
+    using SafeMath for uint256;
+
+    address public constant F_ADDR = 0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF;
+
+    event RequiredSignaturesChanged (uint256 requiredSignatures);
+
+    function setRequiredSignatures(uint256 _requiredSignatures)
+    external
+    onlyOwner
+    {
+        require(validatorCount() >= _requiredSignatures);
+        require(_requiredSignatures != 0);
+        uintStorage[keccak256(abi.encodePacked("requiredSignatures"))] = _requiredSignatures;
+        emit RequiredSignaturesChanged(_requiredSignatures);
+    }
+
+    function getBridgeValidatorsInterfacesVersion()
+    public
+    pure
+    returns (uint64 major, uint64 minor, uint64 patch)
+    {
+        return (2, 1, 0);
+    }
+
+    function requiredSignatures() public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked("requiredSignatures"))];
+    }
+
+    function validatorCount() public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked("validatorCount"))];
+    }
+
+    function isValidator(address _validator) public view returns (bool) {
+        return _validator != F_ADDR && getNextValidator(_validator) != address(0);
+    }
+
+    function isInitialized() public view returns (bool) {
+        return boolStorage[keccak256(abi.encodePacked("isInitialized"))];
+    }
+
+    function deployedAtBlock() public view returns (uint256) {
+        return uintStorage[keccak256("deployedAtBlock")];
+    }
+
+    function getNextValidator(address _address) public view returns (address) {
+        return addressStorage[keccak256(abi.encodePacked("validatorsList", _address))];
+    }
+
+    function validatorList() public view returns (address[]) {
+        address [] memory list = new address[](validatorCount());
+        uint256 counter = 0;
+        address nextValidator = getNextValidator(F_ADDR);
+
+        while (nextValidator != F_ADDR) {
+            list[counter] = nextValidator;
+            nextValidator = getNextValidator(nextValidator);
+            counter++;
+        }
+
+        return list;
+    }
+
+    function deleteItemFromAddressStorage(string _mapName, address _address) internal {
+        delete addressStorage[keccak256(abi.encodePacked(_mapName, _address))];
+    }
+
+    function setValidatorCount(uint256 _validatorCount) internal {
+        uintStorage[keccak256(abi.encodePacked("validatorCount"))] = _validatorCount;
+    }
+
+    function setValidator(address _prevValidator, address _validator) internal {
```

Comment:
Maybe it's better to rename `setValidator` to `setNextValidator`?

Reply:
Thanks for suggestion! Done on https://github.com/poanetwork/poa-bridge-contracts/pull/130/commits/6912fdcad9b8a1c669edd3d0867245984d824e17